src/expressions/functions/builtInFunctions_arrays.ts

```diff
@@ -393,26 +404,123 @@ const arrayForEachPair: FunctionDefinitionType = (
 	);
 };
 
+const isString = (type: string): boolean => {
+	return (
+		isSubtypeOf(type, 'xs:string') ||
+		isSubtypeOf(type, 'xs:anyURI') ||
+		isSubtypeOf(type, 'xs:untypedAtomic')
+	);
+};
+
+const deepLessThan = (
+	dynamicContext,
+	executionParameters,
+	staticContext,
+	valuesA: Value[],
+	valuesB: Value[]
+): boolean => {
+	if (valuesA.length === 0) {
+		return valuesB.length !== 0;
+	} else if (
+		valuesB.length !== 0 &&
+		itemDeepEqual(
+			dynamicContext,
+			executionParameters,
+			staticContext,
+			valuesA[0],
+			valuesB[0]
+		).next(IterationHint.NONE).value
+	) {
+		return deepLessThan(
+			dynamicContext,
+			executionParameters,
+			staticContext,
+			valuesA.slice(1),
+			valuesB.slice(1)
+		);
+	} else if (valuesA[0].value !== valuesA[0].value) {
```

Comment:
this is always false, right?

Reply:
@mehmetscoskun This is a check to see if a value is NaN. Since Nan !== Nan, when valueA is Nan, it returns true (based on w3specs)
`else if ($A[1] ne $A[1] (:that is, $A[1] is NaN:)) 
     then fn:true()`

Reply:
I believe that test-case is already covered by the qt3tests.